Firestore/core/src/model/object_value.h

```diff
@@ -84,15 +84,19 @@ class ObjectValue {
   /**
    * Sets the field to the provided value.
    *
+   * Takes ownership of value.
+   *
    * @param path The field path to set. The path must not be empty.
    * @param value The value to set.
    */
-  void Set(const FieldPath& path, const google_firestore_v1_Value& value);
+  void Set(const FieldPath& path, google_firestore_v1_Value value);
```

Comment:
What do you think about creating a move-only wrapper over protos and using it in function signatures like this?
```cpp
void Set(const FieldPath& path, Unique<google_firestore_v1_Value> value);
// ...
foo.Set(path, std::move(value));
```

The only advantage would be as a sort of documentation -- the caller would have to explicitly move the value, and the fact that ownership is being transferred would be clear in the signature.

Reply:
https://github.com/firebase/firebase-ios-sdk/pull/8244 will replace this argument with `Message`, which should signal the ownership.